pkg/provisioner/ironic/testserver/server.go

```diff
@@ -0,0 +1,104 @@
+package testserver
+
+import (
+	"fmt"
+	"net/http"
+	"net/http/httptest"
+	"testing"
+)
+
+// New returns a MockServer
+func New(t *testing.T, name string) *MockServer {
+	mux := http.NewServeMux()
+	return &MockServer{
+		t:    t,
+		name: name,
+		mux:  mux,
+	}
+}
+
+// MockServer is a simple http testing server
+type MockServer struct {
+	t            *testing.T
+	mux          *http.ServeMux
+	name         string
+	Requests     string
+	FullRequests []*http.Request
+	server       *httptest.Server
+	errorCode    int
+}
+
+// Endpoint returns the URL to the server
+func (m *MockServer) Endpoint() string {
+	if m == nil || m.server == nil {
+		// The consumer of this method expects something valid, but
+		// won't use it if m is nil.
+		return "https://ironic.test/v1/"
+	}
+	response := m.server.URL + "/v1/"
+	m.t.Logf("%s: endpoint: %s", m.name, response)
+	return response
+}
+
+func (m *MockServer) logRequest(r *http.Request, response string) {
+	m.t.Logf("%s: %s %s -> %s", m.name, r.Method, r.URL, response)
+	m.Requests += r.RequestURI + ";"
+	m.FullRequests = append(m.FullRequests, r)
+}
+
+func (m *MockServer) handleNoResponse(w http.ResponseWriter, r *http.Request) {
```

Comment:
On a second thought, I think we could rename it as `handleDefaultResponse`?

Reply:
It looks like this method isn't used at all, so I removed it entirely.